### PR TITLE
Delete Poll Feature 

### PR DESCRIPTION
**Pull request by alexyoung23j**
## Description

Allows admins and thread authors to delete a poll from a thread. 

## Motivation and Context

BD Request. 

## Does this PR affect any server routes?
chose YES or NO

## Have you added the issue number here?
(In the right sidebar, under "development")

### CODE REVIEW

File: packages/commonwealth/server/routes/deletePoll.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      where: {
+        id: thread_id,
+      },
+    });
+
+    if (!thread) throw new AppError(Errors.NoThread);
+    const userOwnedAddressIds = (await req.user.getAddresses())
+      .filter((addr) => !!addr.verified)
+      .map((addr) => addr.id);
+    if (!userOwnedAddressIds.includes(thread.address_id) && !req.user.isAdmin) {
```
Comment (zakhap) [inline packages/commonwealth/server/routes/deletePoll.ts:47]: we need to check community admin, not just siteAdmin

File: packages/commonwealth/server/routes/deletePoll.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  models: DB,
+  req: TypedRequestBody<DeletePollReq>,
+  res: TypedResponse<DeletePollResp>
+) => {
+  if (!req.user) throw new AppError(Errors.NotLoggedIn);
+
+  const { thread_id, poll_id } = req.body;
+  // check chain
+  // check if author of thread
+  // check if admin?
```
Comment (zakhap) [inline packages/commonwealth/server/routes/deletePoll.ts:34]: remove comments

**Review by zakhap — DISMISSED** [review]
🔜

File: packages/commonwealth/server/routes/deletePoll.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+  try {
+    const thread = await models.Thread.findOne({
+      where: {
+        id: thread_id,
+      },
+    });
+
+    if (!thread) throw new AppError(Errors.NoThread);
+    const userOwnedAddressIds = (await req.user.getAddresses())
```
Comment (jnaviask) [inline packages/commonwealth/server/routes/deletePoll.ts:43]: Don't we have a helper for all this admin logic?

File: packages/commonwealth/server/routes/deletePoll.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    const poll = await models.Poll.findOne({
+      where: {
+        thread_id,
+        id: poll_id,
+      },
+    });
+
+    await poll.destroy();
```
Comment (jnaviask) [inline packages/commonwealth/server/routes/deletePoll.ts:68]: What if the poll isn't found? Need to handle case with 400.

File: packages/commonwealth/server/routes/deletePoll.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+    if (!thread) throw new AppError(Errors.NoThread);
+    const userOwnedAddressIds = (await req.user.getAddresses())
+      .filter((addr) => !!addr.verified)
+      .map((addr) => addr.id);
+
+    const userMembership = await findOneRole(
+      models,
+      { where: { address_id: { [Op.in]: userOwnedAddressIds } } },
+      chain_id,
```
Comment (jnaviask) [inline packages/commonwealth/server/routes/deletePoll.ts:50]: Given that this is the only usage of chain_id, the logic of this route means I can use my admin privileges on any community to delete polls anywhere.

**Review by jnaviask — CHANGES_REQUESTED** [review]
Route has critical security issue that needs to be fixed before this is mergeable.

**Comment by CowMuon** [comment]
Given the fairly glaring security concern, converting to draft. (TBH should probably be closed.) 

@alexyoung23j Is this something we can have fixed tomorrow after the Bet1 demo?

File: packages/commonwealth/server/routes/deletePoll.ts (hunk 4 above)
Comment (zakhap) [reply packages/commonwealth/server/routes/deletePoll.ts:68]: fixed this!

File: packages/commonwealth/server/routes/deletePoll.ts (hunk 5 above)
Comment (zakhap) [reply packages/commonwealth/server/routes/deletePoll.ts:50]: Fixed this, pulling from the Thread's chain_id

File: packages/commonwealth/server/routes/deletePoll.ts (hunk 3 above)
Comment (zakhap) [reply packages/commonwealth/server/routes/deletePoll.ts:46]: we have this exact pattern in a dozen + places for getting the user's addresses. i think it's fine for the moment.

**Review by jnaviask — APPROVED** [review]
Created a new poll on a community, then deleted it. Works for me.

**Pull request merged**